src/dune_rules/foreign_rules.ml

```diff
@@ -75,16 +75,21 @@ let include_dir_flags ~expander ~dir (stubs : Foreign.Stubs.t) =
 
 let build_c ~kind ~sctx ~dir ~expander ~include_flags (loc, src, dst) =
   let ctx = Super_context.context sctx in
+  let project = Super_context.find_scope_by_dir sctx dir |> Scope.project in
   let flags =
     let ctx_flags =
       match kind with
       | Foreign_language.C ->
         let cfg = ctx.ocaml_config in
-        List.concat
-          [ Ocaml_config.ocamlc_cflags cfg
-          ; Ocaml_config.ocamlc_cppflags cfg
-          ; Fdo.c_flags ctx
-          ]
+        if Dune_project.new_foreign_flags_handling project then
+          Fdo.c_flags ctx
```

Comment:
Why are FDO flags always passed? is there a test that covers that?

Reply:
They are not always passed, the function may return the empty lsit:
```
let [Fdo.]c_flags (ctx : Context.t) =
  match ctx.fdo_target_exe with
  | None -> []
  | Some _ -> [ "-ffunction-sections" ]
```